Java-Loops.md

```diff
@@ -0,0 +1,208 @@
+
+```java
+do 
+{
+    // Statements
+} 
+while (expression);
+```
+
+This kind of loop is particularly useful if you want your `Statements` to be executed atleast once, irrespective of what `expression` evaluates to. You want to do this if you are initializing a variable inside your loop and plan on using its value later.
+
+```java
+int iter_DoWhile = 20;
+do 
+{
+    System.out.print (iter_DoWhile + " ");
+
+    // Increment the counter
+    iter_DoWhile++;
+} 
+while(iter_DoWhile < 10);
+System.out.println ("iter_DoWhile Value: " + iter_DoWhile);
+```
+
+Output:
+
+    20
+    iter_DoWhile Value: 21
+
+### The For loop
+
+The `for` loop give you a compact way to iterate over a range of values.
+
+```java
+for (initialization; expression; increment) 
+{
+    // Statements
+}
+```
+
+* `initialization` - Initializes the loop and is executed just once, at the beginning.
+* `expression` - Evaluated at the beginning of each iteration. If the `expression` evaluates to `true`, `Statements` will get executed.
+* `increment` - Invoked after each iteration through the loop. You can increase/decrease the value of variables here.
+
+```java
+for (int iter_For = 0; iter_For < 10; iter_For++) 
+{
+    System.out.print (iter_For + " ");
+    // Iterated 10 times, iter_For 0,1,2...9
+}
+System.out.println("iter_For Value: " + fooFor);
+```
+
+Output:
+
+    0 1 2 3 4 5 6 7 8 9
+    iter_For Value: 10
+
+### The For Each loop
+
+Also called the Enhanced For loop, it is an extremely useful and simple way to iterate over each item in a collection, arrays as well as objects that implement the Iterable interface.
+
+```java
+for (object : iterable)
+{
+    // Statements
+}
+```
+
+The loop is read as for each element in the `iterable` (could be an array, collectable etc.). The `object` type must match the element type of the `iterable`. 
+
+```java
+int[] number_list = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+for (int numbers : number_list) 
+{
+    System.out.print (numbers + " ");
+    //Iterated 10 times, numbers 0,1,2...9
+}
+```
+
+Output:
+
+    0 1 2 3 4 5 6 7 8 9
+
+## Loop Control Statemens
+
+### break
+
+Terminates the loop and starts the execution of the code that immediately follows the loop. If you have nested loops, the `break` statement will only end the loop in which it is placed.
+
+```java
+for (int i = 0; i < 10; i++) // Loop 1
+{
+    for (int j = 0; j < 10; j++) // Loop 2
+    {
+        if (i == 5 && j == 5) 
+        {
+            break;
+            // Will terminate Loop 2, but Loop 1 will keep going
+        }
+    }
+}
+```
+
+But if you do want to break out of the outer loop too, you can use a label to exit:
+
+```java
+loop1: // This is a label
+for (int i = 0; i < 10; i++) // Loop 1
+{
+    for (int j = 0; j < 10; j++) // Loop 2
+    {
+        if (i == 5 && j == 5) 
+        {
+            break loop1;
+            // Will break out of Loop 1, instead of just breaking out of Loop 2
+        }
+    }
+}
+```
+
```

Comment:
This is awesome! But if possible, try to think of a real life common problem where you would need this.